aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+
+## How does ASP.NET Core MVC address CSRF?
+
+The most common approach to defending against CSRF attacks is the synchronizer token pattern (STP). STP is a technique where the server sends a token associated with the current user's session to the client with each request that contains a form, and the client must then send back the token to the server for verification. If the server receives a token that doesn't match the authenticated user's session, the request should be rejected. The token is unique and unpredictable, and can also be used to ensure proper sequencing of a series of requests (ensuring page 1 precedes page 2 which precedes page 3). ASP.NET Core MVC will generate Antiforgery Tokens by default on all forms it generates. The following two examples of view logic will generate antiforgery tokens automatically:
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```html
+@using (Html.BeginForm("ChangePassword", "Manage"))
+{
+    
+}
+
+<form asp-controller="Manage" asp-action="ChangePassword" method="post">
+
+</form>
+```
+
+You can also explicitly add an antiforgery token to a ``<form>`` element you create without using tag helpers or HTML helpers by using ``@Html.AntiForgeryToken()``:
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```html
+<form action="/" method="post">
+    @Html.AntiForgeryToken()
+</form>
+```
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```html
+In each of the above cases, ASP.NET Core will add a hidden form field like the following:
+
+<input name="__RequestVerificationToken" type="hidden" value="CfDJ8NrAkSldwD9CpLRyOtm6FiJB1Jr_F3FQJQDvhlHoLNJJrLA6zaMUmhjMsisu2D2tFkAiYgyWQawJk9vNm36sYP1esHOtamBEPvSk1_x--Sg8Ey2a-d9CV2zHVWIN9MVhvKHOSyKqdZFlYDVd69XYx-rOWPw3ilHGLN6K0Km-1p83jZzF0E4WU5OGg5ns2-m9Yw" />
+```
+
+If you don't want to include an antiforgery token as part of your generated form, you can disable the behavior using ``asp-antiforgery="false"``:
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```html
+<form asp-controller="Manage" asp-action="ChangePassword" 
+  method="post" asp-antiforgery="false">
+
+</form>
+```
+
+ASP.NET Core includes three [filters](../mvc/controllers/filters.md) for working with antiforgery tokens: ``ValidateAntiForgeryToken``, ``AutoValidateAntiforgeryToken``, and ``IgnoreAntiforgeryToken``.
```

Comment:
Using this:
`[filters](xref:mvc/controllers/filters)`
Links to filters, but with a bookmark:
`/mvc/controllers/filters.html#mvc_controllers_filters`